ostd/src/arch/x86/boot/smp.rs

```diff
@@ -55,11 +65,28 @@ pub(crate) fn get_num_processors() -> Option<u32> {
 }
 
 /// Brings up all application processors.
-pub(crate) fn bringup_all_aps() {
+pub(crate) fn bringup_all_aps(num_cpus: u32) {
     copy_ap_boot_code();
     fill_boot_stack_array_ptr();
     fill_boot_pt_ptr();
-    send_boot_ipis();
+    cfg_if! {
+        if #[cfg(feature = "cvm_guest")] {
+            if tdx_is_enabled() {
+                for ap_num in 1..num_cpus {
+                    wakeup_aps(
+                        &ACPI_TABLES.get().unwrap().lock(),
+                        AcpiMemoryHandler {},
+                        ap_num,
+                        AP_BOOT_START_PA as u64,
+                        1000,
+                    )
+                    .unwrap();
+                }
+            }
```

Comment:
It seems that if `tdx_is_enabled() == false` and `#[cfg(feature = "cvm_guest")]`, The APs will not be started. I think there misses an `else` that `send_boot_ipis()`.

Reply:
Can non-td machines start APs by ACPI wakeup? If so, what's the hardware feature requirement. Can we just abandon the IPI way in most cases?